src/main/antora/modules/ROOT/partials/mapping.adoc

```diff
@@ -149,6 +149,13 @@ Embedded entities containing a `Collection` or a `Map` will always be considered
 Such an entity will therefore never be `null` even when using @Embedded(onEmpty = USE_NULL).
 endif::[]
 
+[[entity-persistence.embedded-ids]]
+=== Embedded Ids
+
+Entities may be annotated with `@Id` and `@Embedded`, resulting in a composite id on the database side.
```

Comment:
I wonder whether we could simplify our detection. If a simple type is annotated with `@Id`, we consider it a simple identifier. Any non-simple type annotated with `@Id` is automatically considered an embedded/composite Id without the need to use `@Embedded`. `@Embedded` raises the question of `Embedded.onEmpty()`. Being able to provide more detail via `@Embedded` is a nice detail.


It would be also neat to have [Mapping Annotation Overview](https://docs.spring.io/spring-data/mongodb/reference/mongodb/mapping/mapping.html#mapping-usage-annotations) explaining that `@Id` can be simple or embedded/composite identifiers along with embedded Id examples. So far, `@Id` is only used with simple types in code samples.

Reply:
I agree on the documentation part and will tackle that.

I don't like the idea of automatically turning an entity into an embedded when an `@Id` annotation is placed on it. If users want this kind of behavior they could create their own meta annotation.

Also the question of `Embedded.onEmpty()` gets risen by an implicit conversion to an embedded also.